### DISCUSSION TITLE
injected.js No way to use webext-bridge？

### DISCUSSION THREAD

**Discussion by Ya-hui** [wxt-dev/wxt#998] [Q&A]
```ts
// content.ts
export default defineContentScript({
  matches: ['*://www.baidu.com/*'],
  runAt: 'document_start',
  allFrames: true,
  async main() {
    console.log('Injecting script...');
    await injectScript('/injected.js', {
      keepInDom: true,
    });
    console.log('Done!');
  },
});
```
```ts
// injected.ts
import { proxy } from 'ajax-hook';
import { sendMessage } from 'webext-bridge/content-script';
export default defineUnlistedScript(() => {
  console.log('Hello from injected.ts');

  proxy({
    //请求发起前进入
    onRequest: (config, handler) => {
      console.log(config.url);
      handler.next(config);
    },
    //请求发生错误时进入，比如超时；注意，不包括http状态码错误，如404仍然会认为请求成功
    onError: (err, handler) => {
      handler.next(err);
    },
    //请求成功后进入
    onResponse: async (response, handler) => {
      await sendMessage('AJAX-HOOK', response.response, 'background');

      handler.next(response);
    },
  });
});

```
Console prompt Uncaught Error: This script should only be loaded in a browser extension.

> import { sendMessage } from 'webext-bridge/content-script';
> await sendMessage('AJAX-HOOK', response.response, 'background');

What is the reason for this, how to solve very sorry

**Comment by aiktb** [comment]
Umm... You should give a reproducible repo, I don't see what the problem is.

If you delete `sendMessage`, will it still report same error?

**Comment by aklinker1** [comment]
`webext-bridge/content-script` is only for content scripts, not injected scripts that run in the page context. 

It uses the extension API to send messages to the background, it doesn't send messages from an injected script up to the content script. 

So you need to send a message from the injected script to your content script, then use `webext-bridge/content-script` to send the message from your content script to the background.

You can send messages from your injected script to the content script using `window.postMessage`, `CustomEvent`s, or with [externally connectable messaging](https://developer.chrome.com/docs/extensions/develop/concepts/messaging#external-webpage) (recommended).
